robot-command/src/main/java/org/obolibrary/robot/DLQueryCommand.java

```diff
@@ -0,0 +1,174 @@
+      NS + "ILLEGAL RELATION ERROR: %s. Must be one of " + String.join(" ", LEGAL_RELATIONS) + ".";
+  private static final String missingQueryArgumentError =
+      NS + "MISSING QUERY ARGUMENT ERROR: must have a valid --query.";
+
+  /** Error message when --query does not have two arguments. */
+  private static final String missingOutputError =
+      NS + "MISSING OUTPUT ERROR --%s requires two arguments: query and output";
+
+  /** Error message when a query is not provided */
+  private static final String missingQueryError =
+      NS + "MISSING QUERY ERROR at least one query must be provided";
+
+  /** Store the command-line options for the command. */
+  private Options options;
+
+  public DLQueryCommand() {
+    Options o = CommandLineHelper.getCommonOptions();
+    o.addOption("i", "input", true, "load ontology from a file");
+    o.addOption("I", "input-iri", true, "load ontology from an IRI");
+    o.addOption("r", "reasoner", true, "reasoner to use: ELK, HermiT, JFact");
+
+    Option opt = new Option("q", "query", true, "the DL query to run");
+    opt.setArgs(2);
+    o.addOption(opt);
+
+    o.addOption(
+        "s",
+        "select",
+        true,
+        "select what relations to query: equivalents, parents, children, ancestors, descendants, instances");
+    o.addOption("o", "output", true, "save ontology containing only explanation axioms to a file");
+    options = o;
+  }
+
+  @Override
+  public String getName() {
+    return "dl-query";
+  }
+
+  @Override
+  public String getDescription() {
+    return "query the ontology with the given class expression";
+  }
+
+  @Override
+  public String getUsage() {
+    return "robot dl-query --input <file> --query <expression> --output <output>";
+  }
+
+  @Override
+  public Options getOptions() {
+    return options;
+  }
+
+  /**
+   * Handle the command-line and file operations for the DLQueryOperation.
+   *
+   * @param args strings to use as arguments
+   */
+  @Override
+  public void main(String[] args) {
+    try {
+      execute(null, args);
+    } catch (Exception e) {
+      CommandLineHelper.handleException(e);
+    }
+  }
+
+  @Override
+  public CommandState execute(CommandState state, String[] args) throws Exception {
+    CommandLine line = CommandLineHelper.getCommandLine(getUsage(), getOptions(), args);
+    if (line == null) {
+      return null;
+    }
+    if (state == null) {
+      state = new CommandState();
+    }
+    IOHelper ioHelper = CommandLineHelper.getIOHelper(line);
+    state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
+    OWLOntology ontology = state.getOntology();
+
+    OWLReasonerFactory reasonerFactory = CommandLineHelper.getReasonerFactory(line, true);
+    List<String> selects = CommandLineHelper.getOptionalValues(line, "select");
+
+    List<List<String>> queries = getQueries(line);
+    for (List<String> q : queries) {
+      queryOntology(q, ontology, reasonerFactory, selects);
```

Comment:
As implemented, it is very inefficient to run _several_ queries against the same ontology, because there is no shared state between each query (in particular, no shared reasoner, even though the ontology cannot possibly change between each query, so initialising a brand new reasoner instance for each query is pointless).

From some tests I did with Uberon:

* running a single query takes ~11s;
* running 2 queries (`--query "query 1" output1.txt --query "query 2" output2.txt`) takes ~19s;
* running 3 queries takes ~27s;
* and so on.

By contrast, with a modified version that uses a shared state:

* running a single query takes ~11s;
* running 2 queries takes ~11.2s;
* running 3 queries takes ~11.4s;
* and so on.